Running latest golangci-lint (v1.53.3) raises issues with dependencies

The latest version of `golangci-lint` raises issues with `depguard`, e.g.

```sh
cmd/connqc/server.go:10:2: import 'github.com/urfave/cli/v2' is not allowed from list 'Main' (depguard)
        "github.com/urfave/cli/v2"
```

I would suggest to update to the latest `golangci-lint` version and ignore `depguard` issues, as I don't see any benefit in adding an allowlist to our dependencies.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     env:
       GO_VERSION: "1.20"
-      GOLANGCI_LINT_VERSION: v1.51.2
+      GOLANGCI_LINT_VERSION: v1.53.3
 
     steps:
       - name: Check out source code

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -8,6 +8,11 @@ linters-settings:
     skip-tests: true
   gofumpt:
     extra-rules: true
+  depguard:
+    rules:
+      deprecated:
+        deny:
+          - pkg: pkg/errors
 
 linters:
   enable-all: true
@@ -35,7 +40,6 @@ linters:
     - varnamelen
     - wrapcheck
     - wsl
-    - depguard
 
 issues:
   exclude-use-default: false

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -35,6 +35,7 @@ linters:
     - varnamelen
     - wrapcheck
     - wsl
+    - depguard
 
 issues:
   exclude-use-default: false

